src/webgpu/util/f32_interval.ts

```diff
@@ -99,6 +110,33 @@ const kGreaterThanZeroInterval = new F32Interval(
   kValue.f32.positive.max
 );
 
+/** Representation of a vec2/3/4 of floating point intervals as an array of F32Intervals */
+type F32Vector =
+  | [F32Interval, F32Interval]
+  | [F32Interval, F32Interval, F32Interval]
+  | [F32Interval, F32Interval, F32Interval, F32Interval];
+
+/** Coerce F32Interval[] to F32Vector if possible */
+function isF32Vector(v: number[] | F32Interval[]): v is F32Vector {
+  if (v[0] instanceof F32Interval) {
+    return v.length === 2 || v.length === 3 || v.length === 4;
+  }
+  return false;
+}
+
+/** @returns an F32Vector representation of an array fo F32Intervals if possible */
+function toF32Vector(v: number[] | F32Vector): F32Vector {
+  if (isF32Vector(v)) {
+    return v;
+  }
+
+  const f = v.map(toF32Interval);
```

Comment:
map will return a F32Interval[], not a [F32Interval, ...], which is why it needs to be coerced again